libtiledbsoma/src/soma/soma_array.h

```diff
@@ -1157,6 +1157,12 @@ class SOMAArray : public SOMAObject {
             newshape, false, function_name_for_messages);
     }
 
+    /**
+     * This is for SOMADataFrame.
+     */
+    std::pair<bool, std::string> can_upgrade_domain(
+        const ArrowTable& newdomain, std::string function_name_for_messages);
```

Comment:
Can this just reside in the `SOMADataFrame` class or are there issues with doing that, so we need to place it in `SOMAArray`?

Reply:
There are issues with doing that:
https://github.com/single-cell-data/TileDB-SOMA/blob/62dd129515c8e6928d910a2d20f6e0ac431c9fe7/libtiledbsoma/src/soma/soma_array.h#L1230-L1234

I can propagate that comment material

Reply:
Ah ic- thanks. No need to copy it over if it's already in the header.

Reply:
Adding as top-of-file comment :)